What’s up in Vietnam.md

```diff
@@ -1,62 +1,63 @@
+Data Mapper, despite its lower popularity, seems to be a better choice. As we can guess the Data Mapper does not clutter our OODM. Instead, a separated service called EntityManager is mapping the state of our domain objects to a DML and back.
+
+For example:
 
-Example:
 ```
 kate = new User("Kate")
 EntityManager::perist(kate)
 ```
 
-Thanks that we can achieve quite clear domain and hide all dirt behind repositories, but just theoretically. All that process is a little bit magic for most developers. For example, there is a whole document about working with different fetching types in JPA in [this page](https://docs.jboss.org/hibernate/orm/3.6/reference/en-US/html/performance.html#performance-fetching-lazy). If we’re using Hibernate, those fetching types can be overridden by Hibernate specific types if JPA type is EAGER (more [in the documentation](https://docs.jboss.org/hibernate/orm/5.2/userguide/html_single/Hibernate_User_Guide.html#fetching-fetch-annotation)). As a result we are fetching from repositories proxies, not real domain models in some situations, and domain models in others. Default settings are different in different ORM frameworks and different technologies. But it’s a fact that sometimes we gain proxies behind „the repository gate” and we need to remember about it and deal with it.
+Thanks to that we can achieve quite clear domain and hide all dirt behind repositories, but just in theory. They it's being done seems like magic for most developers. For example, there is a huge document about working with different fetching types in JPA on [this page](https://docs.jboss.org/hibernate/orm/3.6/reference/en-US/html/performance.html#performance-fetching-lazy). If you are using Hibernate, those fetching types can be overridden by Hibernate specific types if JPA type is EAGER (more [in the documentation](https://docs.jboss.org/hibernate/orm/5.2/userguide/html_single/Hibernate_User_Guide.html#fetching-fetch-annotation)). As a result we are fetching from repositories' proxies (not real domain models) in some situations and domain models in other cases. Default settings are different in different ORM frameworks and different technologies. But it’s a fact that sometimes we gain proxies behind „the repository gate” and we need to remember about it and deal with it.
 
-As we can see, the use of ORM almost always affects the whole application design. It already sounds awful that the whole application depends on one tool. However, the real hell is waiting behind „the repository gate”. 
+As we can see, the use of ORM almost always affects the application design. It already sounds awful that the whole app should depend on one tool. However, the real hell is waiting behind „the repository gate”.

```

Comment:
1. I wanted to emphasis *whole*. It's the important word here. It's obvious that ORM like every other persistence layer will affect a part of the application design (and application design itself). But it's not obvious that using ORM we need to be careful even in higher abstraction layers such as services or even controllers.

2. IDK if the word `should` is a correct here. It's a fact that `the *whole* application design` depends on ORM and it's a problem.

Reply:
We are leaving the "whole" in the second sentence. With "should" it keeps the same meaning, that it's awful that this is happening.

Reply:
I used word `entire`.

Reply:
I've changed to `I made a fool of myself`.